### DISCUSSION TITLE
Handling Gmail DOM updates in MV3

### DISCUSSION THREAD

**Discussion by hpmayank** [InboxSDK/InboxSDK#1031] [General]
Hello Inbox Team,

I've been investigating the Chrome extension and Inbox SDK for DOM manipulation in Gmail and have a question regarding how DOM updates from Gmail will be managed. Given that manifest version 3 does not support remote code execution, I'm curious about the strategy that the Inbox SDK intends to employ for handling forthcoming Gmail DOM updates.

**Comment by hpmayank** [comment]
Additionally, what's our approach for managing potential scenarios where Gmail opts for a region-specific rollout of the feature that involves updates to the Gmail DOM?

**Comment by aleemstreak** [comment]
For Gmail DOM updates, we have some telemetry in place that can alert us if
the DOM isn't what we expect. Also, given our large user base for Streak we
get users letting us know if things aren't working as expected. Gmail
typically rolls these changes out very very slowly, so we're typically able
to fix any issues before its rolled out to any significant number of
issues. With MV3, you're correct there is no remote code execution - but we
will release updated NPM builds quickly. You'll need to update your
extension and submit for review as soon as we do.
ᐧ

**Reply by hpmayank** [reply]
Thank you for your response, @aleemstreak. Referring to the conversation at https://github.com/InboxSDK/InboxSDK/discussions/1007, I assumed that there could be an early release of Gmail DOM changes provided to the InboxSDK team, and the team will handle any issues prior to the official Gmail DOM update. I'm curious, does the telemetry system cover both addressing existing issues and tracking new DOM changes?
